honeybee_vtk/writer.py

```diff
@@ -0,0 +1,141 @@
+"""Write vtk files to the disk."""
+
+import os
+import json
+import warnings
+
+from zipfile import ZipFile
+from honeybee.typing import clean_string
+from .hbjson import read_hbjson
+from .hbjson import check_grid, get_grid_base, get_grid_mesh, get_grid_points
+from .tovtk import group_by_face_type
+from .tovtk import write_polydata, write_color_grouped_points, write_arrows
+from .helper import get_face_center
+
+
+def write_vtk(
+        file_path, *, file_name=None, get_grids=True,
+        get_vectors=True):
+
+    """Reads a valid HBJSON and writes a .zip of vtk files to the disk.
+
+    Args:
+        file_path: A text string for a valid path to the HBJSON file.
+        file_name: A text string for the name of the .zip file to be written. If no
+            text string is provided, the name of the HBJSON file will be used as a
+            file name for the .zip file.
+        get_grids: A boolean. Defaults to True. Grids will not be extracted from HBJSON
+            if set to False.
+        get_vectors: A boolean. Defaults to True. Vector arrows will not be created if
+            set to False.
+    """
+    # HBJSON file
+    if file_path:
+        # Check if path is fine
+        
+        if not os.path.isfile(file_path):
+            raise FileNotFoundError(
+                'The path is not a valid path.'
+                ' If file exists, try using double backslashes in file path'
+                ' and try again.'
+            )
+    try:
+        # Check if the file is a valid JSON
+        with open(file_path) as fp:
+            hbjson = json.load(fp)
+    except json.decoder.JSONDecodeError:
+        raise ValueError(
+            'Not a valid JSON file.'
+            )
+    else:
+        # Get points and face_types from HBJSON
+        points_lst, hb_types = read_hbjson(hbjson)
+
+        # Get points grouped for each Honeybee face_type
+        grouped_points = group_by_face_type(points_lst, hb_types)
+
+        # Write VTK files based on Honeybee face_type and Honeybee object type
+        for hb_type in grouped_points:
+            write_polydata(grouped_points[hb_type], hb_type)
+
+        # Names of VTK files to be written
+        file_names = list(grouped_points.keys())
+
+        # write grid points to a vtk file if grids are found in HBJSON
+        grids = check_grid(hbjson)
+
+        # If grids are there in HBJSON and they are requested
+        if grids and get_grids:
+
+            # If base_geometry is found
+            if grids[0]:
+                base_geo_points = get_grid_base(grids[0])[0]
+                write_polydata(base_geo_points, 'grid base')
+                file_names.append('grid base')
+
+            # If base_geometry is not found but mesh faces are found
+            if grids[1]:
```

Comment:
shouldn't this be `elif`?

Reply:
No. grids is a tuple of three items.

Reply:
but you want to only generate the 'grid mesh' if there is no 'grid base'. right? Or do you want to add these filenames anyways?

Reply:
It's not just about the file names. Check_grid function returns a tuple of three items. A list of sensorgirds with base_geometry, A list of sensorgrids with mesh and a list of sensorgrids with points. We need to check all three of them and write to file if they are found.

Reply:
Thank you. I think that's the part that I didn't understand. Not knowing about the overall workflow, I think I would change  the code to write the first one that it finds and ignore the rest.

So if grid-base is provided it means we can keep it simple and only write that and save time for generating the rest. If it is not provided then we check the next option and so on but you know the overall process better so I leave it to you.

Reply:
We need to check all three as it often is the case where a Sensorgrid object in HBJSON has different keys. An HBJSON can have all three types of sensorgrids;
1. That has `base_geometry` and `mesh` as keys
2. That has only `mesh` as key
3. That has neither `base_geometry` nor `mesh` as keys
I have already seen files having two cases from above. And I can see that it's very easy to generate an HBJSON that will have all the three above case. That is why I designed this behaviour. I can share some HBJSON files to demo this.